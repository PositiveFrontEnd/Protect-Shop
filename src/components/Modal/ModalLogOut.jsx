import React from "react";
import Modal from "./ModalComponents/Modal";
import ModalWrapper from "./ModalComponents/ModalWrapper";
import ModalHeader from "./ModalComponents/ModalHeader";
import ModalClose from "./ModalComponents/ModalClose";
import ModalBody from "./ModalComponents/ModalBody";
import ModalFooter from "./ModalComponents/ModalFooter";
import SvgLogOut from "./ModalSvg/svgLogOut.svg?react";
import PropTypes from 'prop-types'

const ModalLogOut = ({ isOpen, onclick, firstClick, secondaryClick }) => {
    return (
        <>
            <ModalWrapper isOpen={isOpen}>
                <Modal>
                    <ModalHeader>
                        <div className="header__title">
                            <SvgLogOut />
                            <p className="title__text">Donâ€™t leave us!</p>
                        </div>
                        <ModalClose onclick={onclick} />
                    </ModalHeader>
                    <ModalBody>
                        <p className="modal__body-title title__logOut">do you want to log out?</p>
                        <ModalFooter
                            className="footer__email"
                            secondaryText="Yes"
                            firstText="No"
                            firstClassName="modal__button"
                            secondaryClassName="modal__button"
                            firstClick={firstClick}
                            secondaryClick={secondaryClick}
                        />
                    </ModalBody>
                </Modal>
            </ModalWrapper>
        </>
    )
}

ModalLogOut.defaultProps = {
    onclick: () => { },
    isOpen: () => { }
}

ModalLogOut.propTypes = {
    onclick: PropTypes.func,
    isOpen: PropTypes.func,
}

export default ModalLogOut