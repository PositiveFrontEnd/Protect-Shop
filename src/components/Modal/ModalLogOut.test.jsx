import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import ModalLogOut from "./ModalLogOut";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  ModalLogOut", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };
        const data = {
            onclick: () => { },
            isOpen: () => { },
            secondaryClick: () => { },
            firstClick: () => { },
        };
        const store = mockStore(initialState);
        const modalLogOut = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ModalLogOut data={data} />
                </BrowserRouter>
            </Provider>
        );

        expect(modalLogOut).toMatchSnapshot();
    });
});