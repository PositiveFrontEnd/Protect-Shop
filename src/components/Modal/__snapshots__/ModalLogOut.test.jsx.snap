// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  ModalLogOut > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal__wrapper"
      >
        <div
          class="modal"
        >
          <div
            class="modal__header"
          >
            <div
              class="header__title"
            >
              <svg
                class="header__svg"
                fill="none"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  id="Group"
                >
                  <g
                    id="Group_2"
                  >
                    <path
                      d="M8 0C3.58171 0 0 3.58171 0 8C0 12.4183 3.58171 16.0001 8 16.0001C12.4181 16.0001 16 12.4183 16 8C16 3.58171 12.4181 0 8 0ZM3.79462 7.26721C3.96176 7.08811 4.24023 7.07659 4.42169 7.2461C4.9017 7.69687 5.39456 7.28183 5.44919 7.2331C5.61234 7.08693 5.8539 7.08368 6.02015 7.21553C6.03979 7.23089 6.0584 7.24846 6.07567 7.2678C6.23897 7.4503 6.22347 7.73069 6.04097 7.89413C5.64763 8.24613 4.68141 8.70385 3.81558 7.89413C3.63648 7.72685 3.62733 7.44631 3.79462 7.26721ZM11.3111 12.4661C11.0865 12.5636 10.826 12.4601 10.7289 12.2355C10.3014 11.2479 9.29224 10.6096 8.15651 10.6096C6.99494 10.6096 5.97926 11.2474 5.56923 12.2346C5.49851 12.4047 5.33358 12.5078 5.16009 12.5078C5.10354 12.5078 5.04596 12.4968 4.99044 12.4738C4.76453 12.3798 4.65749 12.1205 4.75125 11.8947C5.29918 10.5759 6.63586 9.72382 8.15666 9.72382C9.64571 9.72382 10.9746 10.5716 11.542 11.8838C11.639 12.1082 11.5356 12.3688 11.3111 12.4661ZM12.1691 7.88897C11.7757 8.24097 10.8095 8.69868 9.94367 7.88897C9.76487 7.72168 9.75571 7.44114 9.92271 7.26204C10.0897 7.08294 10.3685 7.07143 10.5498 7.24093C11.0298 7.69171 11.5226 7.27666 11.5773 7.22794C11.7404 7.08176 11.982 7.07851 12.1482 7.21036C12.1679 7.22572 12.1865 7.24329 12.2038 7.26263C12.3672 7.44513 12.3517 7.72552 12.1691 7.88897Z"
                      fill="#FEFEFE"
                      id="Vector"
                    />
                  </g>
                </g>
              </svg>
              <p
                class="title__text"
              >
                Don’t leave us!
              </p>
            </div>
            <button
              class="header__button"
              type="button"
            >
              <svg
                fill="black"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                  fill="white"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="modal__container"
          >
            <div
              class="modal__content modal__content-form"
            >
              <p
                class="modal__body-title title__logOut"
              >
                do you want to log out?
              </p>
              <div
                class="footer__email"
              >
                <button
                  class="button-white modal__button"
                  type="button"
                >
                  Yes
                </button>
                <button
                  class="button-black modal__button"
                  type="button"
                >
                  No
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal__wrapper"
    >
      <div
        class="modal"
      >
        <div
          class="modal__header"
        >
          <div
            class="header__title"
          >
            <svg
              class="header__svg"
              fill="none"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Group"
              >
                <g
                  id="Group_2"
                >
                  <path
                    d="M8 0C3.58171 0 0 3.58171 0 8C0 12.4183 3.58171 16.0001 8 16.0001C12.4181 16.0001 16 12.4183 16 8C16 3.58171 12.4181 0 8 0ZM3.79462 7.26721C3.96176 7.08811 4.24023 7.07659 4.42169 7.2461C4.9017 7.69687 5.39456 7.28183 5.44919 7.2331C5.61234 7.08693 5.8539 7.08368 6.02015 7.21553C6.03979 7.23089 6.0584 7.24846 6.07567 7.2678C6.23897 7.4503 6.22347 7.73069 6.04097 7.89413C5.64763 8.24613 4.68141 8.70385 3.81558 7.89413C3.63648 7.72685 3.62733 7.44631 3.79462 7.26721ZM11.3111 12.4661C11.0865 12.5636 10.826 12.4601 10.7289 12.2355C10.3014 11.2479 9.29224 10.6096 8.15651 10.6096C6.99494 10.6096 5.97926 11.2474 5.56923 12.2346C5.49851 12.4047 5.33358 12.5078 5.16009 12.5078C5.10354 12.5078 5.04596 12.4968 4.99044 12.4738C4.76453 12.3798 4.65749 12.1205 4.75125 11.8947C5.29918 10.5759 6.63586 9.72382 8.15666 9.72382C9.64571 9.72382 10.9746 10.5716 11.542 11.8838C11.639 12.1082 11.5356 12.3688 11.3111 12.4661ZM12.1691 7.88897C11.7757 8.24097 10.8095 8.69868 9.94367 7.88897C9.76487 7.72168 9.75571 7.44114 9.92271 7.26204C10.0897 7.08294 10.3685 7.07143 10.5498 7.24093C11.0298 7.69171 11.5226 7.27666 11.5773 7.22794C11.7404 7.08176 11.982 7.07851 12.1482 7.21036C12.1679 7.22572 12.1865 7.24329 12.2038 7.26263C12.3672 7.44513 12.3517 7.72552 12.1691 7.88897Z"
                    fill="#FEFEFE"
                    id="Vector"
                  />
                </g>
              </g>
            </svg>
            <p
              class="title__text"
            >
              Don’t leave us!
            </p>
          </div>
          <button
            class="header__button"
            type="button"
          >
            <svg
              fill="black"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                fill="white"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal__container"
        >
          <div
            class="modal__content modal__content-form"
          >
            <p
              class="modal__body-title title__logOut"
            >
              do you want to log out?
            </p>
            <div
              class="footer__email"
            >
              <button
                class="button-white modal__button"
                type="button"
              >
                Yes
              </button>
              <button
                class="button-black modal__button"
                type="button"
              >
                No
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
